src/components/ModalWithConfirm.js

```diff
@@ -0,0 +1,20 @@
+import Modal from "./Modal";
+
+export default class ModalWithConfirm extends Modal {
+  constructor(modalSelector) {
+    super({modalSelector});
+    this._form = this._modalElement.querySelector(".modal__form");
+  }
+
+  handleDelete(handleDeleteSubmit) {
+    this._handleDeleteSubmit = handleDeleteSubmit;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._form.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleDeleteSubmit();
+    });
+  }
+}
```

Comment:
Needs correction: please add `renderLoading` method for rendering loading (you can copy it from ModalWithForm)